gridstatus/isone.py

```diff
@@ -127,12 +149,26 @@ def get_fuel_mix(self, date, end=None, verbose=False):
 
         df["Date"] = pd.to_datetime(df["Date"] + " " + df["Time"])
 
-        # groupby FuelCategory to make it possible to infer DST changes
+        logger.debug(f"Looking for DST transition on {date}...")
+        is_dst_transition = (
+            df["Date"].dt.date == pd.Timestamp("2024-11-03").date()
+        ) & (df["Date"].dt.hour == 1)
+        transition_idx = None
+
+        if is_dst_transition.any():
+            logger.debug(
+                "Found DST transition. Looking for fallback transition time...",
+            )
+            time_diffs = df["Date"].diff()
+            negative_diffs = time_diffs[time_diffs < pd.Timedelta(minutes=-30)]
```

Comment:
i see the issue. in theory this approach of looking at the order is what pandas `ambiguous=infer` does. any idea why it stopped working? 

like if you try to run the original code on DST transition during november last year, it infers correctly

<img width="991" alt="image" src="https://github.com/user-attachments/assets/6aba052b-efe0-4f39-aae2-4e56e1284f16">

Reply:
Hmm, interesting, yeah not sure why it stopped working then. Especially since it gives an `AmbiguousTimeError` which one would think a "here's what you do when it's ambiguous?" kwarg would cover. 🤷🏼

Reply:
Load, LMP, solar and wind forecasts use the same ambiguous="infer" argument and they correctly parsed the transition yesterday. I wonder what makes the fuel mix different?

Reply:
Ohh, the problem is the "Other" fuel category doesn't have any repeats. Here's the data

<img width="616" alt="Screenshot 2024-11-04 at 11 23 01" src="https://github.com/user-attachments/assets/513f28f8-3d4c-47a7-861a-b061ef1e6698">

Compare to "Hydro" (or any other Fuel Category) that has repeats.

<img width="586" alt="Screenshot 2024-11-04 at 11 24 06" src="https://github.com/user-attachments/assets/bac87da5-f6e9-4ae3-969d-af9734099442">

Reply:
I think we can have the goal of moving all applicable datasets to the API by the next DST transition. [Here's a ticket to capture that](https://www.notion.so/gridstatus/Swap-ISO-New-England-methods-for-API-134e835f42aa802a9443e0ca2c5addbf?pvs=4).

Reply:
`ambiguous="infer"` works on all fuel categories except for "Other". I think we can simplify the fix:

```python
groups = []

# groupby Fuel Category to make it possible to infer DST changes
for _category, group in df.groupby("Fuel Category"):
    try:
        group["Date"] = group["Date"].dt.tz_localize(
            self.default_timezone,
            ambiguous="infer",
        )
    # Some groups ("Other") may not have duplicate times and thus Pandas cannot
    # infer the correct timezone. In these cases, assume the timestamps
    # are the first occurrence
    except pytz.AmbiguousTimeError:
        group["Date"] = group["Date"].dt.tz_localize(
            self.default_timezone, ambiguous=True
        )

    groups.append(group)

df = pd.concat(groups)
```

Reply:
swapping for the API